utils/bazel/llvm-project-overlay/llvm/config.bzl

```diff
@@ -2,118 +2,331 @@
-    "HAVE_STRERROR_R=1",
-    "HAVE_SYSEXITS_H=1",
-    "HAVE_UNISTD_H=1",
-]
-
-linux_defines = posix_defines + [
-    "_GNU_SOURCE",
-    "HAVE_MALLINFO=1",
-    "HAVE_SBRK=1",
-    "HAVE_STRUCT_STAT_ST_MTIM_TV_NSEC=1",
-]
-
-macos_defines = posix_defines + [
-    "HAVE_MACH_MACH_H=1",
-    "HAVE_MALLOC_MALLOC_H=1",
-    "HAVE_MALLOC_ZONE_STATISTICS=1",
-    "HAVE_PROC_PID_RUSAGE=1",
-    "HAVE_UNW_ADD_DYNAMIC_FDE=1",
-]
-
-win32_defines = [
-    # Windows system library specific defines.
-    "_CRT_SECURE_NO_DEPRECATE",
-    "_CRT_SECURE_NO_WARNINGS",
-    "_CRT_NONSTDC_NO_DEPRECATE",
-    "_CRT_NONSTDC_NO_WARNINGS",
-    "_SCL_SECURE_NO_DEPRECATE",
-    "_SCL_SECURE_NO_WARNINGS",
-    "UNICODE",
-    "_UNICODE",
+CONFIG_H_SUBSTITUTIONS = (
+    {"${BUG_REPORT_URL}": "https://github.com/llvm/llvm-project/issues/"} |
+    cmakedefine01(
+        "ENABLE_BACKTRACES",
+        disable = "//llvm/config:LLVM_ENABLE_BACKTRACES_disabled",
+    ) |
+    cmakedefine01(
+        "ENABLE_CRASH_OVERRIDES",
+        disable = "//llvm/config:LLVM_ENABLE_CRASH_OVERRIDES_disabled",
+    ) |
+    cmakedefine01(
+        "LLVM_ENABLE_CRASH_DUMPS",
+        enable = "//llvm/config:LLVM_ENABLE_CRASH_DUMPS_enabled",
+    ) |
+    cmakedefine01(
+        "ENABLE_DEBUGLOC_COVERAGE_TRACKING",
+        enable = "//llvm/config:LLVM_ENABLE_DEBUGLOC_COVERAGE_TRACKING_coverage",
+    ) |
+    cmakedefine01(
+        "LLVM_WINDOWS_PREFER_FORWARD_SLASH",
+        enable = (
+            "@bazel_tools//tools/cpp:mingw",
+            "//llvm/config:LLVM_WINDOWS_PREFER_FORWARD_SLASH_enabled",
+        ),
+        disable = (
+            "//llvm/config:LLVM_WINDOWS_PREFER_FORWARD_SLASH_disabled",
+            "//conditions:default",
+        ),
+    ) |
+    cmakedefine("HAVE_BACKTRACE", enable = "//llvm/config:posix") |
+    {"${BACKTRACE_HEADER}": "execinfo.h"} |
+    {
+        "#cmakedefine HAVE_CRASHREPORTERCLIENT_H": "/* #undef HAVE_CRASHREPORTERCLIENT_H */",
+    } |
+    cmakedefine01_off("HAVE_CRASHREPORTER_INFO") |
+    cmakedefine01_off("HAVE_DECL_ARC4RANDOM") |
+    cmakedefine01_on("HAVE_DECL_FE_ALL_EXCEPT") |
+    cmakedefine01_on("HAVE_DECL_FE_INEXACT") |
+    cmakedefine01_off("HAVE_DECL_STRERROR_S") |
+    cmakedefine_set("HAVE_DLOPEN") |
+    cmakedefine("HAVE_REGISTER_FRAME", enable = "//llvm/config:posix") |
+    cmakedefine("HAVE_DEREGISTER_FRAME", enable = "//llvm/config:posix") |
+    cmakedefine("HAVE_UNW_ADD_DYNAMIC_FDE", enable = "@platforms//os:macos") |
+    cmakedefine_unset("HAVE_FFI_CALL") |
+    cmakedefine_unset("HAVE_FFI_FFI_H") |
+    cmakedefine_unset("HAVE_FFI_H") |
+    cmakedefine_set("HAVE_FUTIMENS") |
+    cmakedefine_set("HAVE_FUTIMES") |
+    cmakedefine_set("HAVE_GETPAGESIZE") |
+    cmakedefine_set("HAVE_GETRUSAGE") |
+    {"#cmakedefine HAVE_ISATTY 1": "#define HAVE_ISATTY 1"} |
+    cmakedefine_unset("HAVE_LIBEDIT") |
+    # TODO: Wire this up Exegesis and remove custom logic.
+    cmakedefine_unset("HAVE_LIBPFM") |
```

Comment:
Note: Libpfm is already configurable but doesn't match the CMake behavior. By disabling it here we keep the existing behavior for now and can defer changes to the libpfm configuration.